src/expire.c

```diff
@@ -229,22 +244,25 @@ void activeExpireCycle(int type) {
          * we scanned. The percentage, stored in config_cycle_acceptable_stale
          * is not fixed, but depends on the Redis configured "expire effort". */
         do {
-            unsigned long num, slots;
+            unsigned long num;
             iteration++;
 
             /* If there is nothing to expire try next DB ASAP. */
-            if ((num = dictSize(db->expires)) == 0) {
+            if ((num = dbSize(db, EXPIRE_DICT)) == 0) {
                 db->avg_ttl = 0;
                 break;
             }
-            slots = dictSlots(db->expires);
             data.now = mstime();
 
-            /* When there are less than 1% filled slots, sampling the key
+            /* When there are less than 1% filled buckets, sampling the key
              * space is expensive, so stop here waiting for better times...
              * The dictionary will be resized asap. */
-            if (slots > DICT_HT_INITIAL_SIZE &&
-                (num*100/slots < 1)) break;
+            if (!server.cluster_enabled) {
+                unsigned long buckets = dictSlots(db->expires[0]);
+                if (buckets > DICT_HT_INITIAL_SIZE && (num*100/buckets < 1)) {
+                    break;
+                }
```

Comment:
it's a little hard for me to follow the code here (probably due to using small monitor with GH, and being too lazy to clone the code).

i have a feeling something is odd here, why is this code not needed in cluster mode?
can we somehow avoid duplicating this logic and big comment?